server/server.go

```diff
@@ -469,43 +469,81 @@ func New(
 	return s, nil
 }
 
-func (s *Server) getHostIP(configIP string) net.IP {
+func (s *Server) getHostIPs(configIP string) []string {
 	// emulate kubelet behavior of choosing hostIP
 	// ref: k8s/pkg/kubelet/nodestatus/setters.go
+	rootIP := ""
 
 	// use configured value if set
 	if hostIP := net.ParseIP(configIP); hostIP != nil {
-		return hostIP
+		rootIP = hostIP.String()
 	}
 
 	// Otherwise use kubernetes utility to choose hostIP
 	// there exists the chance for both hostIP and err to be nil, so check both
-	if hostIP, err := knet.ChooseHostInterface(); err != nil && hostIP != nil {
-		return hostIP
+	if rootIP == "" {
+		if hostIP, err := knet.ChooseHostInterface(); err != nil && hostIP != nil {
```

Comment:
All ChooseHostInterface does is pick the interface with the default route. If v4 is not configured on the machine, it will return a v6 address.

Reply:
I added neccesary IPv6 paths.